ircii: fix build on older systems and Sonoma

barracuda156 opened the pull request:
#### Description

Fix

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 10.6
Xcode 3.2

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

pmetzger commented:
What's the failure on '14?

pmetzger commented on irc/ircii/Portfile:
```diff
@@ -36,6 +36,17 @@ configure.env       COPY_DIRECTORY=tar
+# from incompatible pointer type [-Wincompatible-pointer-types]
+if {[string match *gcc* ${configure.compiler}]} {
+    configure.cflags-append \
+                    -Wno-error=incompatible-pointer-types
```
As usual, I note that upstream probably wants to know about such problems.

barracuda156 replied:
Homepage returns “The page you requested could not be found”.

pmetzger replied:
Try talking to mrg a t netbsd.org

ryandesign replied:
The current homepage might be http://eterna23.net/ircii/

barracuda156 replied:
“ircII comes with a script called ircbug that should be used to send bug reports”

Sorry, I am not gonna do this. If everyone prefers having this port broken, so be it, it is not important for me. There is more important stuff in the pipeline to complete.

barracuda156 commented:
`clang-18` builds it on Sonoma with no hacks. I will just blacklist Apple clang there.

pmetzger commented:
FYI, I've updated the underlying port, and I've emailed Matt.

barracuda156 committed:
ircii: fix build with gcc

barracuda156 commented:
> FYI, I've updated the underlying port, and I've emailed Matt.

Thank you. I rebased to master now. (Both issues with gcc are still there, so fixes are due.)

pmetzger commented:
Why don't we tell him about the warning? I bet he'll just fix it.

reneeotten commented:
so, this is still failing. If the issue is only with GCC we could in theory merge this as most people will use Clang anyway. But the CI for macOS 14 is still not passing - what's the path forward?

barracuda156 commented:
> so, this is still failing. If the issue is only with GCC we could in theory merge this as most people will use Clang anyway. But the CI for macOS 14 is still not passing - what's the path forward?

@reneeotten It was building fine with clang-18 on Sonoma. It is an easy fix: avoid a broken Apple clang.

reneeotten commented:
then please do so...

barracuda156 committed:
ircii: avoid Apple clang on Sonoma, use LLVM

pmetzger commented:
If we know exactly what's wrong, I can get Matt to take patches.

pmetzger commented:
Matt has a new upstream version, I'm going to update to it so that we can have a new default server again.

reneeotten commented:
poke @pmetzger @barracuda156 let's get this wrapped up plase or close the PR.

barracuda156 commented:
@reneeotten Since apparently @pmetzger wants to do the update, let’s hope he could also borrow compiler blacklist to have older systems fixed.
In the meanwhile I close this.
